recipes/functionalplus/all/conanfile.py

```diff
@@ -0,0 +1,62 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class FunctionalPlusConan(ConanFile):
+    name = "functionalplus"
+    description = "Functional Programming Library for C++."
+    license = "BSL-1.0"
+    topics = ("conan", "functionalplus", "fplus", "functional programming")
+    homepage = "https://github.com/Dobiasd/FunctionalPlus"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "4.9",
+            "Visual Studio": "14",
+            "clang": "3.7",
+            "apple-clang": "9",
+        }
+
+    def configure(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 14)
+
+        def lazy_lt_semver(v1, v2):
+            lv1 = [int(v) for v in v1.split(".")]
+            lv2 = [int(v) for v in v2.split(".")]
+            min_length = min(len(lv1), len(lv2))
+            return lv1[:min_length] < lv2[:min_length]
+
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if not minimum_version:
+            self.output.warn("functionalplus requires C++14. Your compiler is unknown. Assuming it supports C++14.")
+        elif lazy_lt_semver(str(self.settings.compiler.version), minimum_version):
```

Comment:
What's the reason you are not using https://github.com/conan-io/conan/blob/d83046a4faa2789817f8136a6d088bf524d0e50d/conans/client/tools/version.py#L11 ?

It's a full semver compare 🤔 I would expect it to be valid here

Reply:
No, if someone has only major version in its profile (which is pretty common), il might raise an invalid configuration error (gcc 4 in profile for example, because we compare with 4.9).

Reply:
There's over 100 instance of this blob of code in CCI, now you got me worried about how many others might be impacted!

Reply:
Some of them

Reply:
@prince-chrismc @SpaceIm IMO if you encounter repeating code pattern in CCI about 100 times, for me it's a clear candidate to implement it as a conan helper or tool. otherwise, it would be a maintenance nightmare to fix copy-pasted code in 100 places.

Reply:
I think we have 2 of these so far... it's just the 50-75 recipes with the compiler check might require this... 

💯 we need a tool for this and theres already an issue in the main repo (you opened as I recall)